### PR TITLE
LPS-124257 Create test to check if the date mask is applied according to the locale

### PR DESCRIPTION
**Pull request by alinedoleron**
Issue related: https://issues.liferay.com/browse/LPS-124257

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -78,17 +56,39 @@ const getDelimiter = (dateFormat) => {
...
+const getLocaleDateFormat = (format = 'L') => {
+	return moment.localeData().longDateFormat(format);
+};
+
+const getMaskByDateFormat = (format) => {
+	const mask = [];
+
+	for (let i = 0; i < format.length; i++) {
+		if (/[a-z]/i.test(format[i])) {
+			mask.push(/\d/);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:68]: @alinedoleron, can you please add the regular expressions (`/[a-z]/i` and `/\d/`) to constants to let the code a bit clear?

Besides, is this code the same used on Workflow? If so, maybe you and the Workflow guys should consider create a util to share this code among different modules. What the Workflow guys think about it?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 2]
```diff
@@ -78,17 +56,39 @@ const getDelimiter = (dateFormat) => {
...
+		}
+	}
+
+	return mask;
+};
+
 const getDateFormat = () => {
-	const dateFormat = Liferay.AUI.getDateFormat();
-	const dateDelimiter = getDelimiter(dateFormat);
+	const dateMask = getLocaleDateFormat();
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:79]: @alinedoleron, why did you rename this variable? Shouldn't it still be named `dateFormat`?

Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:79]: @alinedoleron, why did you rename this variable? Shouldn't it still be named `dateFormat`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js (hunk 1 above)
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:68]: @alinedoleron, can you please add the regular expressions (`/[a-z]/i` and `/\d/`) to constants to let the code a bit clear?

Besides, is this code the same used on Workflow? If so, maybe you and the Workflow guys should consider create a util to share this code among different modules. What the Workflow guys think about it?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 3]
```diff
@@ -78,17 +56,39 @@ const getDelimiter = (dateFormat) => {
...
-		dateMask: getDateMask(dateFormat, dateDelimiter),
-		inputMask: getInputMask(dateFormat, dateDelimiter),
+		dateMask: getDateMask(dateMask, dateDelimiter),
+		inputMask,
 	};
 };
 
-const transformToDate = (date) => {
+const transformToDate = (date, locale) => {
+	moment.locale(locale);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:90]: @alinedoleron, shouldn't this line be inside the `if`? After all, the moment is only used inside of it.

Additionally, can we do something like `moment(date).locale(locale).toDate();`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 4]
```diff
@@ -199,7 +201,8 @@ const DatePicker = ({
 					}
 
 					if (moment(value).isValid()) {
-						onChange(moment(value).format(dateMask.toUpperCase()));
+						moment.locale(locale);
+						onChange(moment(value).format('MM/DD/YYYY'));
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:205]: Additionally, can we do something like `moment(value).locale(locale).format('MM/DD/YYYY');`?

By the way, can you please clarify why are you using this hard coded format (MM/DD/YYYY)?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 5]
```diff
@@ -194,4 +194,36 @@ describe('DatePicker', () => {
 
 		expect(onChange).toHaveBeenCalledWith({}, date);
 	});
+
+	it('fills the input with the current date according to the locale', async () => {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:198]: Nice job, @alinedoleron! 😄

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js (hunk 4 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:205]: @natocesarrego, this was Carol who indicated to do, because this is the way back end needs to receive. Without it the date selected blinks. For example, if I select a date in pt_BR, 15/03/20, at first it shows correctly and then it changes to 03/15/20.

**Comment by alinedoleron** [comment]
new pr: https://github.com/natocesarrego/liferay-portal/pull/941

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js (hunk 4 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:205]: Thanks for your clarification, @alinedoleron! 😉
